### PR TITLE
Feat/pickup slot caps

### PR DESCRIPTION
**Pull request by BanerjeeAgniva**
# Pull Request

> **Scope / merge-order note.** Because the stacked branches were merged into their parents, this branch (`feat/pickup-slot-caps`) now carries **two** features, and neither is on `main` yet — so this PR's diff is **both**:
> 1. per-slot pickup capacity cap (**#11**) — *also* PR #151, and
> 2. per-merchant section categories (**#1**).
>
> **Recommended:** merge **#151 first** (slot caps → main); this PR's diff then shrinks to **categories only** and can merge as a clean single-feature PR. Alternatively, close #151 and merge this one to land both features in one go.

## Summary
Two merchant/booking features landing on `main`: (1) a **per-slot pickup capacity cap** that rejects a booking whose pickup slot is full *before payment*, and (2) **per-merchant section categories** — a B2B merchant defines their own categories, tags a shipment at booking, and filters/reports by them.

---

## What Changed

**Per-merchant section categories (#1)**
- `merchant_category` table (Flyway `V4_41`) + `shipments.category_id` (nullable, tagged at booking).
- Account-scoped CRUD `/api/v1/categories` (`MerchantCategoryController/Service/Repository`) — unique name per merchant, case-insensitive.
- `B2bBookingRequest.categoryId` stamped after **same-merchant validation** (a shipment can only carry one of the booking account's own categories).
- `category_id` surfaced on the customer shipment read model (`MyShipmentSummaryResponse`).
- `MerchantCategoryServiceImplTest` over create / duplicate / rename / delete scoping.

**Per-slot pickup capacity cap (#11)** — same as #151, included here because stacked
- `PickupSlotCapacity` enforced pre-payment in both booking flows; full/invalid slot → `PickupSlotFullException` → **HTTP 409**.
- Ops-tunable `orders.pickup-slot.max-per-slot` (default 50); only `DA_PICKUP` shipments count (SELF_DROP excluded).

---

## Why This Change?
- **Categories:** merchants organise shipments into their own sections (e.g. "Electronics", "Apparel") and filter/report by them — the platform previously had no per-merchant grouping.
- **Slot cap:** a pickup window has finite DA capacity on a one-day SLA; checking before payment means a full slot rejects cleanly instead of charging then failing.

---

## Testing
- [x] Tested locally (`mvn test -pl orders`)
- [x] Edge cases considered (category same-merchant validation, duplicate names; ASAP / self-drop / invalid-hour for the slot cap)
- [x] No breaking changes introduced

### Test Evidence
```
mvn -pl orders -am test -Dtest=MerchantCategoryServiceImplTest,PickupSlotCapacityTest,\
  BookingServiceImplTest,B2bBookingServiceImplTest,AdminOrderQueryServiceExportForAccountTest,OrderStatusReducerTest
...
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
BUILD SUCCESS
```
Also verified end-to-end on localhost: the categories CRUD + tagging drive the business console, and the slot cap returns 409 on a full window.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java [hunk 1]
```diff
@@ -194,6 +197,10 @@ public BookingResponse book(BookingRequest req, String idempotencyKey, String us
         int chargeableWeightGrams = priced.chargeableWeightGrams();
         QuoteResult quote = priced.quote();
 
+        // ── 3b. Pickup-slot capacity (before payment, so a full slot never charges) ──
+        pickupSlotCapacity.ensureRoom(req.getOriginCity(), req.getPickupSlotDate(),
+                req.getPickupSlotStartHour(), req.getPickupType());
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java:202]: Valid observation. The cart/bulk checkout path (`CartServiceImpl` → `bookSettled`) is intentionally not slot-capped in this slice — only the single B2C/B2B `book()` paths are. Unlike single bookings, cart checkout captures one aggregate payment upfront for N items, so a per-item pre-payment check raises an unsettled product question (reject the whole cart if one item’s slot is full, or partially book?). Rather than bake that policy in here, single-booking capping ships now and cart-flow capping is a tracked follow-up — not silently dropped.

File: orders/src/main/java/com/oneday/orders/service/impl/MerchantCategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    @Transactional
+    public MerchantCategoryResponse create(UUID accountId, MerchantCategoryRequest request) {
+        String name = request.getName().trim();
+        requireUnique(accountId, name);
+        MerchantCategory entity = new MerchantCategory();
+        entity.setB2bAccountId(accountId);
+        entity.setName(name);
+        return MerchantCategoryResponse.from(repository.save(entity));
+    }
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/MerchantCategoryServiceImpl.java:40]: The 500 premise doesn’t hold in this module: `OrdersGlobalExceptionHandler` already maps `DataIntegrityViolationException` → **409 CONFLICT**, so the concurrent-duplicate race returns 409 (the unique index `ux_merchant_category_account_name` is the backstop and it’s translated), not 500. Catching it in the service to rethrow `DuplicateCategoryException` would make the race return the same specific message as the pre-check path — a nice consistency touch I’m happy to add as a small follow-up, but there’s no 500 exposure today.

File: orders/src/main/java/com/oneday/orders/service/impl/PickupSlotCapacity.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        if (pickupType != PickupType.DA_PICKUP) {
+            return;   // self-drop consumes no DA pickup slot
+        }
+        String city = originCity.toUpperCase();
+        Instant slotStart = PickupSlots.resolve(slotDate, startHour).start();
+        int booked = shipmentRepository
+                .countByOriginCityAndScheduledPickupStartAndPickupTypeAndCancelledAtIsNull(
+                        city, slotStart, PickupType.DA_PICKUP);
+        if (booked >= props.getMaxPerSlot()) {
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/PickupSlotCapacity.java:58]: Deliberate soft cap for v1 — flagged in code with a `ponytail:` comment naming the ceiling and the upgrade path. The count-then-book window can overbook a slot by up to (concurrent bookings − 1). A pickup-slot cap is an operational capacity hint, not a hard invariant like payment, and mild overbooking under rare concurrency self-corrects operationally, so we chose not to pay for a per-slot lock/counter at pilot volume. The upgrade path (per-slot reservation row + row lock, mirroring `VanManifestServiceImpl.bindEarliest`) is documented in-code and can be swapped in if overbooking actually bites. Leaving open as a tracked, accepted tradeoff.

File: orders/src/main/resources/db/migration/orders/V4_41__create_merchant_category.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+CREATE UNIQUE INDEX ux_merchant_category_account_name ON merchant_category (b2b_account_id, lower(name));
+
+CREATE TRIGGER trg_merchant_category_updated_at
+  BEFORE UPDATE ON merchant_category
+  FOR EACH ROW EXECUTE FUNCTION set_updated_at();
+
+-- Tag a shipment with one of its merchant's categories (null = untagged). FK-by-convention to
+-- merchant_category; set at booking time, so immutable like the other booking facts.
+ALTER TABLE shipments ADD COLUMN category_id UUID;
+CREATE INDEX idx_shipments_category ON shipments (b2b_account_id, category_id);
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/resources/db/migration/orders/V4_41__create_merchant_category.sql:24]: Fair best-practice, deferring for now. `merchant_category` is a brand-new empty table so its indexes build instantly, and `idx_shipments_category` on the existing shipments table builds sub-second at pilot volume, so the transactional SHARE lock is negligible. `CREATE INDEX CONCURRENTLY` also requires a separate non-transactional Flyway migration (can’t run inside the migration’s tx) — worth doing once shipments volume makes the build time material, tracked as a follow-up rather than added now.

File: orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/BookingServiceImpl.java:202]: create issue

File: orders/src/main/java/com/oneday/orders/service/impl/PickupSlotCapacity.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/java/com/oneday/orders/service/impl/PickupSlotCapacity.java:58]: make issue

File: orders/src/main/resources/db/migration/orders/V4_41__create_merchant_category.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply orders/src/main/resources/db/migration/orders/V4_41__create_merchant_category.sql:24]: make issue
